ddtrace/tracer/option.go

```diff
@@ -1209,6 +1209,16 @@ func ChildOf(ctx ddtrace.SpanContext) StartSpanOption {
 	}
 }
 
+func ChildOfWithExtractedSpanLinks(ctx ddtrace.SpanContext) StartSpanOption {
+	return func(cfg *ddtrace.StartSpanConfig) {
+		if spanCtx, ok := ctx.(*spanContext); ok && spanCtx != nil {
+			cfg.SpanLinks = spanCtx.spanLinks
+			spanCtx.spanLinks = nil
+		}
+		cfg.Parent = ctx
+	}
+}
```

Comment:
I feel this function does too much. Also, `ChildOf` is deprecated in `v2`, so I would suggest reducing the scope and changing the name for this function to something like this:

```suggestion
func WithSpanLinks(ctx ddtrace.SpanContext) StartSpanOption {
	return func(cfg *ddtrace.StartSpanConfig) {
		if spanCtx, ok := ctx.(*spanContext); ok && spanCtx != nil {
			cfg.SpanLinks = spanCtx.spanLinks
			// TODO: explain why this side effect is required
			spanCtx.spanLinks = nil
		}
	}
}
```

Reply:
@mhlidd Thanks for updating the name.